_posts/2016/2016-06-23-karlsruher-entwicklertag.md

```diff
@@ -0,0 +1,90 @@
+Over time, lots of manual synchronisation errors between annotated refactored test classes in the source code and the test scenarios may occur, especially in larger projects.
+Here is where JGiven shows its strength.
+JGiven starts with the code, just runs the tests, and outputs a report, and a pending (TODO) test list.
+By using a defined business-readable test case syntax a great readability in the source code itself could be achieved, and even a product owner is still able to write tests.
+Furthermore, it features Spring Auto-Wiring, stage abstraction, quantity parameters, and test object abstraction allocation.
+Other competitive frameworks are Spock (in Groovy) and Serenity (in Java).
+
+### "JUnit5", by Johannes Link and Matthias Merdes
+
+{% image blog/blog-karlsruhe-junit.jpg %}
+
+Interestingly, the next major iteration of [JUnit 5](http://junit.org/junit5/) raised over 53,000 euros on [IndieGogo](https://www.indiegogo.com/#/picks_for_you), which could finance several arrangements like a workshop in Karlsruhe with guys from the big IDE companies, eclipse and IntelliJ, as well as Pivotal for an optimised tooling support.
+The runner concept from the previous version is abolished so that you can use multiple extensions in parallel.
+
+Another interesting feature is `@assertAll`, which does multiple asserts without stopping when in the meantime a failure occurs, so that the test result is more significant (e.g. a comparison of all personal data including addresses).
+What else?
+
+* Annotations can now be used for tags or descriptions.
+* You may also create your own test annotations with retention policy hooks.
+* Nested tests allow BDD-style test case structures.
+* All test are executed dynamically without a runner, which show the results progressively.
+
+### "Better Java-Code with Type Annotations)", by Dragan Zuvic
+
+{% image blog/blog-karlsruhe-jsr.jpg %}
+
+A talk that presented many conceptions on how to write your own types and format checkers.
+Quite interesting!
+
+### Keynote: "Microservices", by Eberhard Wolff
+
+{% image blog/blog-karlsruhe-microservices.jpg %}
+
+Key topics of this talk: how to deal with maintainability, redundant data, and redundant code for software systems.
+
+Eberhard showed in several iterations how legacy code was tried to be rewritten, and later ended in another legacy code project even so new methods and ideas were applied.
+Albert Einstein calls this "Insanity".
+
+Doing the same thing over and over again and expecting different results is a dilemma that the software industry has not been able to solve yet.
+Eric Evans' approach for domain driven design with an ubiquitous language, value objects etc. does often not work but introduced the helpful concept of bounded contexts.
```

Comment:
does often not work for solving this architectural erosion problem

architekturelle erosion ist ein feststehender fachbegriff